src/rgw/rgw_rest_pubsub_common.cc

```diff
@@ -10,6 +10,13 @@
 #define dout_context g_ceph_context
 #define dout_subsys ceph_subsys_rgw
 
+bool verify_transport_security(CephContext *cct, const RGWEnv& env) {
+  if (g_conf().get_val<bool>("rgw_allow_secrets_in_cleartext")) {
+    return true;
+  }
+  return rgw_transport_is_secure(cct, env);
```

Comment:
can we please add a scary warning message at log level 0 whenever this override is triggered?

Reply:
added a warning here: https://github.com/ceph/ceph/pull/43436/files#diff-7dd5f6f1e1de636d2699c66e02f38ca4e255a48f14304f8ad54b6c68e9248e94R15